cli/internal/taskhash/taskhash.go

```diff
@@ -94,8 +94,8 @@ func (pfs *packageFileSpec) hash(pkg *fs.PackageJSON, repoRoot fs.AbsolutePath)
 	return hashOfFiles, nil
 }
 
-func manuallyHashPackage(pkg *fs.PackageJSON, inputs []string, rootPath fs.AbsolutePath) (map[string]string, error) {
-	hashObject := make(map[string]string)
+func manuallyHashPackage(pkg *fs.PackageJSON, inputs []string, rootPath fs.AbsolutePath) (map[turbopath.RelativeUnixPath]string, error) {
```

Comment:
I like the switch to the typed return here, and I agree that this is the right type for hashing.

Reply:
Yeah, it's fixing these method signatures (and then plumbing through the consequences of that) that really make a difference for correctness.

My comments on Friday (while frustrated, I'm sorry, I wanted to get this landed, but each revision makes it better) about this were focused on traveling the path from "I changed the method signature" to "I have types that go through the entire system. It's that interim state when I didn't know *which* type I wanted where, and where changing one meant a cascade up to *multiple* different call sites which may not be doing the right thing (as demonstrated by #1413).

All of the changes I've made with this have resulted in better code, except for in my opinion c27caff8073868eee9e62a5fce2784badf8de098 which just feels like pulling the ladder up after you've climbed the tree.